clang/test/SemaCXX/local-class-template-param-crash.cpp

```diff
@@ -0,0 +1,14 @@
+// RUN: %clang_cc1 -fsyntax-only %s -verify
+// expected-no-diagnostics
+
+template <int i>
+int g() {
+  return [] (auto) -> int {
+    struct L {
+      int m = i;
+    };
+    return 0;
+  } (42);
```

Comment:
I feel like this is some fallout of not decoupling instantiating lambda body from the lambda transformation. If we had been able to avoid the body transform when transforming the lambda for the call expression, we could also have avoided instantiating an offending intermediate `L`, right? @erichkeane

Reply:
That seems about right. 
But the fix is probably fine? Maybe we can add a FIXME in the comment for now.

Reply:
Yeah, probably also worth an assertion for which we’re within a lambda.